client/components/Carousel.jsx

```diff
@@ -0,0 +1,92 @@
+/* eslint-disable react/no-array-index-key */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import Card from './Card.jsx';
+import CarouselModal from './CarouselModal.jsx';
+
+class Carousel extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      modalActive: false,
+      selectedIndex: 0,
+    };
+  }
+
+  prevSlide = () => {
+    const { selectedIndex } = this.state;
+    const newIndex = selectedIndex > 0 ? selectedIndex - 1 : 0;
+    this.setState({
+      selectedIndex: newIndex,
+    });
+  }
+
+  nextSlide = () => {
+    const { selectedIndex } = this.state;
+    const { photos } = this.props;
+    const newIndex = selectedIndex < photos.length - 1 ? selectedIndex + 1 : photos.length - 1;
+    this.setState({
+      selectedIndex: newIndex,
+    });
+  }
+
+  clickShowModal = () => {
+    this.setState({
+      modalActive: true,
+    });
+  }
+
+  clickHideModal = () => {
+    this.setState({
+      modalActive: false,
+    });
+  }
```

Comment:
Maybe toggleModal will further simplify this.

Reply:
Good idea! I'll make a note to return to this in the future.